src/js/components/LoginPage/Login.js

```diff
@@ -1,11 +1,100 @@
 import React, { Component } from "react";
 import "./Login.scss";
 import "./ButtonStyles.css";
+import Navbar from "../Navbar/Navbar";
+import {Redirect} from "react-router-dom";
+import StyledFirebaseAuth from "../RegisterPage/SignUp";
+import {RegisterUser} from "../../services/apiservice";
+import { BrowserRouter, Route, Switch, NavLink } from "react-router-dom";
+
+
 class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      surname: '',
+      email: '',
+      phone: '',
+      password: '',
+      organisation: '',
+
+    };
+
+    // this.handleClick = this.handleClick.bind(this);
+  }
+
+  updateNameValue(evt) {
+    this.setState({name: evt.target.value});
+  }
+
+  updateSurnameValue(evt) {
+    this.setState({surname: evt.target.value});
+  }
+
+  updateEmailValue(evt) {
+    this.setState({email: evt.target.value});
+  }
+
+  updatePhoneValue(evt) {
+    this.setState({phone: evt.target.value});
+  }
+
+  updatePasswordValue(evt) {
+    this.setState({password: evt.target.value});
+  }
+  updateOrganisationValue(evt) {
+    this.setState({organisation: evt.target.value});
+  }
+
+  handleClick() {
```

Comment:
Why aren't you binding the method in the constructor?

Reply:
When googling this was how they suggested I do the binding, with the current version of react that my team is using.

Reply:
Strive to explicitly bind your method in the constructor so that `this` can surely be bound to the class method once the component is initialized.
That will help you to not run in cases where `this.handleClick` is undefined when you're calling the method.

Reply:
Alright thank you I will look into it, and I will communicate it to my team because many seniors value unity and as a team we unitiedly all agreed to use this methodology.
Thank you so much for the advice.